### PR TITLE
Upgrade axe-core to 3.5.3

### PR DESCRIPTION
**Pull request by benbcai**
### Summary
Upgrade axe-core to 3.5.3
Closes https://github.com/cerner/terra-toolkit-boneyard/issues/12
### Additional Details
- Upgrade axe-core to 3.5.3 that includes the fix for https://github.com/cerner/terra-toolkit/issues/318
- This will be a MVB
- Added upgrade guide
- Enable the following new axe rules since they are not causing any failures in Terra components. Continue to disable `scrollable-region-focusable`.
  - `aria-input-field-name`
  - `aria-roledescription`
  - `aria-toggle-field-name`
  - `avoid-inline-spacing`
  - `input-button-name`
  - `landmark-unique`
  - `role-img-alt`

### Test Runs
https://github.com/cerner/terra-core/pull/2930
https://github.com/cerner/terra-clinical/pull/646
https://github.com/cerner/terra-framework/pull/1093
https://github.com/cerner/terra-application/pull/28
https://github.com/cerner/terra-dev-site/pull/251
https://github.cerner.com/orion/orion-dev-site-plugin/pull/6
https://github.cerner.com/orion/orion-component-example-js/pull/74
https://github.cerner.com/orion/dynamic-component-catalog/pull/71
https://github.cerner.com/orion/dynamic-component-catalog-toolkit/pull/19

@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,13 @@ Changelog
 
 Unreleased
 ----------
+### Major Version Bump.
```
Comment (emilyrohrbough) [inline CHANGELOG.md:6]: Are we ready for this MVB?

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,13 @@ Changelog
...
+
+### Fixed
+* Fixed wdio test termination caused by axe-core error in IE when inspecting a page that contains an SVG element.
```
Comment (emilyrohrbough) [inline CHANGELOG.md:9]:
```suggestion
* Fixed wdio test termination caused by axe-core v3.2.0 error in IE when inspecting a page that contains an SVG element.
```

File: docs/guides/UpgradeGuide-v6.0.0.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Terra Toolkit Upgrade Guide v4.0.0
```
Comment (emilyrohrbough) [inline docs/guides/UpgradeGuide-v6.0.0.md:1]:
```suggestion
# Terra Toolkit Upgrade Guide v6.0.0
```

File: src/wdio/services/TerraCommands/chai-methods.js [hunk 4]
```diff
@@ -8,15 +8,11 @@ function accessible() {
   // eslint-disable-next-line no-underscore-dangle
   new chai.Assertion(this._obj).to.be.instanceof(Array);
 
-  // These are the new rules introduced in axe-core v3.3 and v3.4 to disable for now to remain passive.
+  /**
+    * This rules was introduced in axe-core v3.3 and causes failures in many Terra components. 
+    * The solution to address this failure vary by component. It is being disabled until a solution is identified in the future.
```
Comment (emilyrohrbough) [inline src/wdio/services/TerraCommands/chai-methods.js:13]: Is there an issue we can reference?

File: CHANGELOG.md (hunk 1 above)
Comment (mjhenkes) [reply CHANGELOG.md:6]: Yes, we want to fix IE testing internally. We're going to limit the blast radius by allowing multiple major version in dev-site, orion-toolkit, and that mpage fusion toolkit.

**Comment by emilyrohrbough** [comment]
@benbcai can we open up a few test PRs for this

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,13 @@ Changelog
 
 Unreleased
 ----------
+### Major Version Bump.
+
+### Fixed
+* Fixed wdio test termination caused by axe-core v3.2.0 error in IE when inspecting a page that contains an SVG element.
+
+### Changed
```
Comment (StephenEsser) [inline CHANGELOG.md:11]: We're a little inconsistent, but I think ***Breaking Changes*** stands out a little more in the change log

```
### Breaking Changes
* Upgrade axe-core version from 3.2.0 to 3.5.3
```

File: src/wdio/services/TerraCommands/chai-methods.js [hunk 6]
```diff
@@ -8,15 +8,11 @@ function accessible() {
   // eslint-disable-next-line no-underscore-dangle
   new chai.Assertion(this._obj).to.be.instanceof(Array);
 
-  // These are the new rules introduced in axe-core v3.3 and v3.4 to disable for now to remain passive.
+  /**
+    * This rules was introduced in axe-core v3.3 and causes failures in many Terra components.
```
Comment (StephenEsser) [inline src/wdio/services/TerraCommands/chai-methods.js:12]: `This rules was` -> `These rules were`

or

`This rules was` -> `This rule was`

Depending on if we're treating these as singular or plural

File: src/wdio/services/TerraCommands/chai-methods.js (hunk 4 above)
Comment (benbcai) [reply src/wdio/services/TerraCommands/chai-methods.js:13]: Added the logged issue for reference: https://github.com/cerner/terra-toolkit-boneyard/pull/24/commits/e2d7e126b6d247e6774158a794550c343d20db57

File: CHANGELOG.md (hunk 5 above)
Comment (benbcai) [reply CHANGELOG.md]: Updated: https://github.com/cerner/terra-toolkit-boneyard/pull/24/commits/001f8e60a0625c85e2ad185b6a282af07796b921

File: src/wdio/services/TerraCommands/chai-methods.js (hunk 6 above)
Comment (benbcai) [reply src/wdio/services/TerraCommands/chai-methods.js]: I updated the previous comment but didn't update the word `rules`. https://github.com/cerner/terra-toolkit-boneyard/pull/24/commits/7966c02bd85fd5bc44a18820f6313c7f529211f3

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,13 @@ Changelog
 
 Unreleased
 ----------
+### Major Version Bump.
+
+### Fixed
+* Fixed wdio test termination caused by axe-core v3.2.0 error in IE when inspecting a page that contains an SVG element.
+
+### Breaking Change
+* Upgrade axe-core version from 3.2.0 to 3.5.3
```
Comment (emilyrohrbough) [inline CHANGELOG.md:12]: Can we move this to the top of the Change log ?

Comment (benbcai) [reply CHANGELOG.md:9]: Updated: https://github.com/cerner/terra-toolkit-boneyard/pull/24/commits/8c5825068fc6f990c8fd86328fa91cb33ff8ab45

**Pull request merged**